crates/swc_ecma_codegen/tests/fixture/issue-6589/js/output.js

```diff
@@ -0,0 +1,14 @@
+export function func({ a , b , ...rest }) {
+    console.log(a, b, rest);
+}
+const other = {
+    unknow: "foo"
+};
+let foo = {
+    ...other,
+    bar: "baz"
+};
+let bar = {
+    bar: "baz",
+    ...other
```

Comment:
ES2022 allows trailing comma here, but we don't have option to force enable/disable it (or we can print it when it supported, but currently we don't use this logic), so I think we should solve it in the separated PR